pkg/operators/wasm/maps.go

```diff
@@ -236,7 +245,19 @@ func (i *wasmOperatorInstance) mapUpdate(ctx context.Context, m wapi.Module, sta
 		return
 	}
 
-	err = ebpfMap.Update(key, value, ebpfFlags)
+	if getFD {
+		targetMapHandle := binary.NativeEndian.Uint32(value)
+		targetMap, ok := getHandle[*ebpf.Map](i, targetMapHandle)
+		if !ok {
+			stack[0] = 1
+			return
+		}
+
+		err = ebpfMap.Update(key, uint32(targetMap.FD()), ebpfFlags)
```

Comment:
Could you check if `ebpfMap` is a map that accepts a fd? (`BPF_MAP_TYPE_PROG_ARRAY`, `BPF_MAP_TYPE_ARRAY_OF_MAPS`, `BPF_MAP_TYPE_HASH_OF_MAPS`, ...)

Otherwise, if the `ebpfMap` is a regular `BPF_MAP_TYPE_HASH`, this would provide a way for the wasm module to discover the fd of maps.

Additionally, the check will be helpful for gadget author to debug problems.

Similar in the "else" case, please check that `ebpfMap` does not accept a fd.